xla/hlo/evaluator/hlo_evaluator.cc

```diff
@@ -318,11 +324,11 @@ std::optional<ParamIndexAndValue> TryParsingInstructionAsParameterAndInteger(
   }
   std::optional<DynamicOrStaticInteger> integer_value =
       GetInstructionValueAsInteger(instruction, precomputed_analyses);
-  result.value = std::move(integer_value);
+  result.value = integer_value;
   if (!result.IsValid()) {
     return std::nullopt;
   }
-  return std::optional<ParamIndexAndValue>(std::move(result));
+  return result;
```

Comment:
- removing std::move because ParamIndexAndValue is trivially copyable.
- removing std::optional too because wrapping result with std::optional is done automatically.